File: Python Program/Count Number of digits in a number.py
# defining the function
def num_digit(n):
    count=0
    # while loop to iterate over the values
    while(n>0):
        # counting the digits
        count=count+1
        n=n//10
    # returing the count
    return count

# displaying the result and function calling
print(num_digit(123))
print(num_digit(1234))
print(num_digit(67812))

File: Java Program/Java Recursion/Count Number of digits in a number.java
// importing the libraries
import java.util.*;
import java.io.*;
// main class
class Main{
    // declaring the main method
	public static void main(String[] args){
		Scanner sc = new Scanner(System.in);
		// getting input from user
		int n = sc.nextInt();
		if(n>=0 && n<=999999){
		    int count=0;
    		// interating over the digits
    		while (n != 0) {
    			n = n / 10;
    			count+=1;
    		}
    		// displaying the result
    		System.out.println(count);
		}else{
		    System.out.println("Out of range");
		}
	}
}

